### DISCUSSION TITLE
Hubert ssl

### DISCUSSION THREAD

**Discussion by pratikkumar018** [espnet/espnet#3712] [General]
I am trying to run ssl with hubert according to this configuration file. 
https://github.com/espnet/espnet/blob/master/egs2/librispeech/ssl1/conf/tuning/train_asr_hubert_base_960h_pretrain_it2.yaml

When I set skip_nomask=true, I get divide by zero error as count_unmask =0 (https://github.com/espnet/espnet/blob/master/espnet2/hubert/espnet_model.py line 253). If I set skip_nomask=false, everything runs fine. Is this the expected behavior or when we set skip_nomask=true, it should skip the count of unmask ?

**Comment by sw005320** [comment]
@simpleoier, can you answer it?

**Comment by simpleoier** [comment]
Hi @pratikkumar018 . Thanks for reporting this. This is a bug. When `skip_nomask=true`, the `count_unmask` would be 0. The division to compute the accuracy would raise the error. I will quickly fix this.

**Reply by pratikkumar018** [reply]
Does it affect the Pre-training? We are trying to pretrain the model initialized with hubert_ll60k and on a different dataset. Should it be set to false or true?

**Reply by simpleoier** [reply]
It should affect the performance. This `skip_nomask` determines whether to care about the predictions in the unmasked regions, which is used [here](https://github.com/pytorch/fairseq/blob/265df7144c79446f5ea8d835bda6e727f54dad9d/fairseq/models/hubert/hubert.py#L493). TBH, I'm not sure how much it will affect.
